pkg/controllers/management/nodepool/nodepool.go

```diff
@@ -56,15 +62,73 @@ func (c *Controller) Create(nodePool *v3.NodePool) (runtime.Object, error) {
 	return nodePool, nil
 }
 
+func (c *Controller) needsReconcile(nodePool *v3.NodePool) bool {
+	changed, _, err := c.createOrCheckNodes(nodePool, true)
+	if err != nil {
+		logrus.Debugf("[nodepoool] error checking pool for reconciliation: %s", err)
+	}
+
+	return changed
+}
+
+func (c *Controller) reconcile(nodePool *v3.NodePool) {
+	_, qty, err := c.createOrCheckNodes(nodePool, false)
+	if err != nil {
+		logrus.Errorf("[nodepool] reconcile erorr, create or check nodes: %s", err)
+	}
+
+	var np *v3.NodePool
+	if qty != nodePool.Spec.Quantity {
+		// pool was scaled down, need to update qty
+		np = nodePool.DeepCopy()
+		np.Spec.Quantity = qty
+	} else {
+		np = nodePool
```

Comment:
why do we need to update node pool if the quantity isn't changed?

Reply:
To clear the annotation